stan/math/prim/fun/positive_ordered_constrain.hpp

```diff
@@ -18,21 +18,16 @@ namespace math {
  * @param x Free vector of scalars.
  * @return Positive, increasing ordered vector.
  */
-template <typename T>
-Eigen::Matrix<T, Eigen::Dynamic, 1> positive_ordered_constrain(
-    const Eigen::Matrix<T, Eigen::Dynamic, 1>& x) {
-  using Eigen::Dynamic;
-  using Eigen::Matrix;
+template <typename Vec, require_vector_like_t<Vec>* = nullptr>
+inline auto positive_ordered_constrain(Vec&& x) {
   using std::exp;
-  using size_type = index_type_t<Matrix<T, Dynamic, 1>>;
-
-  size_type k = x.size();
-  Matrix<T, Dynamic, 1> y(k);
+  auto k = x.size();
+  plain_type_t<Vec> y(k);
   if (k == 0) {
     return y;
   }
   y[0] = exp(x[0]);
-  for (size_type i = 1; i < k; ++i) {
+  for (auto i = 1; i < k; ++i) {
     y[i] = y[i - 1] + exp(x[i]);
   }
   return y;
```

Comment:
`return cumulative_sum(exp(x))`?

Reply:
I need to look into this but that was giving me the test failure

```
test/unit/math/prim/fun/positive_ordered_transform_test.cpp:25: Failure
Expected equality of these values:
  exp(1.0) + exp(-2.0) + exp(-5.0)
    Which is: 2.86036
  y[2]
    Which is: 2.86036
```

Reply:
Well the test uses `EXPECT_EQ` for floating point numbers. It would fail for anything that is not exactely the same expression. The test should be modified to have some tolerance.